src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/request/RaptorRequestTransferCache.java

```diff
@@ -66,6 +68,23 @@ public RaptorRequestTransferCache(int maximumSize) {
                 ))
                 .values()))
             .collect(toList());
+
+        var reversedTransfers = new ArrayList<List<RaptorTransfer>>(forwardTransfers.size());
+
+        for (int i = 0; i < forwardTransfers.size(); i++) {
+            reversedTransfers.add(new ArrayList<>());
+        }
+
+        for (int fromStop = 0; fromStop < forwardTransfers.size(); fromStop++) {
+            final int finalFromStop = fromStop;
+            forwardTransfers.get(fromStop)
+                    .forEach(transfer -> {
+                        reversedTransfers.get(transfer.stop())
+                                .add(new ReversedRaptorTransfer(finalFromStop, transfer));
+                    });
+        }
+
+        return new RaptorTransferIndex(forwardTransfers, reversedTransfers);
```

Comment:
Could this be replaced with on iteration over the transfers putting each transfer into the 2 resulting maps using a "old fashion" loop?

Reply:
I've moved the method into `RaptorTransferIndex` and simplified the logic a bit.